SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialTypebindingManager.cpp

```diff
@@ -28,169 +30,208 @@ void USpatialTypebindingManager::Init(USpatialNetDriver* InNetDriver)
+	{
+		int32 Offset = SubobjectClassDataPair.Key;
+		FSubobjectSchemaData SubobjectSchemaData = SubobjectClassDataPair.Value;
+
+		FSoftClassPath SubobjectClassPath(SubobjectSchemaData.ClassPath);
+		UClass* SubobjectClass = SubobjectClassPath.ResolveClass();
+		if (SubobjectClass == nullptr)
+		{
+			continue;
+		}
+
+		FClassInfo* SubobjectInfoPtr = FindClassInfoByClass(SubobjectClass);
+		if (SubobjectInfoPtr == nullptr)
+		{
+			continue;
 		}
 
+		// Make a copy of the already made FClassInfo for this specific subobject
+		TSharedPtr<FClassInfo> ActorSubobjectInfo = MakeShared<FClassInfo>(*SubobjectInfoPtr);
+		ActorSubobjectInfo->SubobjectName = SubobjectSchemaData.Name;
+
 		ForAllSchemaComponentTypes([&](ESchemaComponentType Type)
 		{
-			Worker_ComponentId ComponentId = SchemaDatabase->ClassPathToSchema[Class->GetPathName()].SchemaComponents[Type];
+			Worker_ComponentId ComponentId = SubobjectSchemaData.SchemaComponents[Type];
 			if (ComponentId != 0)
 			{
-				Info.SchemaComponents[Type] = ComponentId;
-				ComponentToClassMap.Add(ComponentId, Class);
-				ComponentToOffsetMap.Add(ComponentId, 0);
-				ComponentToCategoryMap.Add(ComponentId, (ESchemaComponentType)Type);
+				ActorSubobjectInfo->SchemaComponents[Type] = ComponentId;
+				ComponentToClassInfoMap.Emplace(ComponentId, ActorSubobjectInfo);
+				ComponentToOffsetMap.Add(ComponentId, Offset);
+				ComponentToCategoryMap.Add(ComponentId, ESchemaComponentType(Type));
 			}
 		});
 
-		Info.Class = Class;
+		// This is needed because the recursive call to FindClassInfoByClass for subobjects changes the map
+		Info = ClassInfoMap.FindChecked(Class);
+		Info.SubobjectInfo.Add(Offset, ActorSubobjectInfo);
+	}
+}
 
-		ClassInfoMap.Emplace(Class, Info);
+UClass* USpatialTypebindingManager::LoadClassForComponent(Worker_ComponentId ComponentId)
+{
+	// Only try to load classes for generated components
+	if (ComponentId < SpatialConstants::STARTING_GENERATED_COMPONENT_ID)
+	{
+		return nullptr;
 	}
 
-	for (UClass* Class : SupportedClasses)
+	for (auto& ClassDataPair : SchemaDatabase->ClassPathToSchema)
 	{
-		for (auto& SubobjectDataPair : SchemaDatabase->ClassPathToSchema[Class->GetPathName()].SubobjectData)
+		for (int32 Type = SCHEMA_Begin; Type < SCHEMA_Count; Type++)
 		{
-			int32 Offset = SubobjectDataPair.Key;
-			FSubobjectSchemaData SubobjectSchemaData = SubobjectDataPair.Value;
-
-			FClassInfo* ActorInfo = FindClassInfoByClass(Class);
-
-			FSoftClassPath SubobjectClassPath(SubobjectSchemaData.ClassPath);
-			UClass* SubobjectClass = SubobjectClassPath.TryLoadClass<UObject>();
-			if (SubobjectClass == nullptr)
-			{
-				continue;
-			}
-
-			FClassInfo* SubobjectInfoPtr = FindClassInfoByClass(SubobjectClass);
-			if (SubobjectInfoPtr == nullptr)
+			const Worker_ComponentId ObjectComponentId = ClassDataPair.Value.SchemaComponents[Type];
+			if (ComponentId == ObjectComponentId)
 			{
-				continue;
+				FSoftClassPath SoftClassPath(ClassDataPair.Key);
```

Comment:
This code could be lifted out into a helper function as the same pattern is used in three places.